### PR TITLE
feature:generate environments on index file and upload to s3

### PR DESCRIPTION
**Pull request by jonmediajel**
https://github.com/MediaJel/mediajel-tracker/issues/420

- CI WORKFLOW is still TODO

Needed ENV

```
# Generate Environment Variables
ASK DEVOPS or JON for ENV Variables
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,5 +56,8 @@
   },
   "browserslist": [
     "since 2017-06"
-  ]
+  ],
+  "dependencies": {
+    "aws-sdk": "^2.1643.0"
+  }
```
Comment (pacholoamit) [inline package.json:62]: THis is a dev dependency. This will get bundled in our tracker

File: public/index.html [hunk 2]
```diff
@@ -66,6 +66,68 @@
...
+      'send_to': 'AW-11413419653/p0DZCPaMq_cYEIXtq8Iq',
+      'value': parseFloat(invoice.totalAmount),
+      'currency': 'USD',
+      'transaction_id': retrievedObject.orderNo.toString()});
+      sessionStorage.setItem("pixelData", "0");
+    }
+  } catch (error) {}
+});
+
+</script>
```
Comment (pacholoamit) [inline public/index.html:130]: I think this shouldn't be included in your PR

File: src/scripts/generateEnvironments.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    if (writeErr) {
+      console.error('Error writing environment.js file:', writeErr);
+      return;
+    }
+    console.log('environment.js file created successfully!');
+
+    // Upload the file to S3
+    uploadToS3(environmentPath, bucketName, s3Key);
+  });
+});
```
Comment (pacholoamit) [inline src/scripts/generateEnvironments.ts:86]: Looks bueno

**Review by pacholoamit — CHANGES_REQUESTED** [review]
Minor changes

File: package.json (hunk 1 above)
Comment (jonmediajel) [reply package.json]: Duly noted, thank you

File: public/index.html (hunk 2 above)
Comment (jonmediajel) [reply public/index.html]: Thanks for the feedback

**Comment by elmobunani21** [comment]
> #420
> 
> * CI WORKFLOW is still TODO
> 
> Needed ENV
> 
> ```
> # Generate Environment Variables
> AWS_ACCESS_KEY_ID_PROD=""
> AWS_SECRET_ACCESS_KEY_PROD=""
> AWS_REGION_PROD=us-east-1
> BUCKET_GENERATE_ENVIRONMENTS_PROD=mjcarts
> ```

Pls remove the access credentials for security purpose, AWS will detect as security risk.

**Pull request merged**